i3_workspace_names_daemon.py

```diff
@@ -39,21 +38,55 @@ def build_rename(i3, app_icons, delim):
         The rename callback.
     """
     def get_icon_or_name(leaf):
-        window_class = leaf.window_class.lower()
+        window_class = ""
+        if leaf.window_class:
+            window_class = leaf.window_class.lower()
+        else:
+            window_class = leaf.name.lower()
+
         if window_class in app_icons and app_icons[window_class] in icons:
             return icons[app_icons[window_class]]
         else:
             return window_class.lower()
 
     def rename(i3, e):
-        for workspace in i3.get_tree().workspaces():
+        workspaces = i3.get_tree().workspaces()
+        #need to use get_workspaces since the i3 con object doesn't have the visible property for some reason
+        workdicts = i3.get_workspaces()
+        visible =  [workdict['name'] for workdict in workdicts if workdict['visible']]
+        visworkspaces = []
+        focus = [workdict['name'] for workdict in workdicts if workdict['focused']]
```

Comment:
True, albeit a little hard to read in my opinion. Will change.

Reply:
@cboddy I see you updated your comment. Would you prefer you're way or mine? I personally would opt for mine since I try not to import when it can be avoided.

Reply:
@kensand NP, yours looks good.

Reply:
Uh oh... my way is causing focus to be None incorrectly for some reason, I'll try you're way